### PR TITLE
Refactors Robotic Emotes

### PR DESCRIPTION
**Pull request by Ziiro**
:cl: 
fixes: Borgs can emote again
tweak: Drones can use robotic emotes and scream
/:cl:

Emote code is a fucking mess, but this makes robotic emotes work for robotic entities

### CODE REVIEW

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -98,6 +113,9 @@
 
 /datum/emote/proc/can_run_emote(mob/user, help_check)
 	. = TRUE
+	if(robotic_emote == TRUE)
```
Comment (FlattestGuitar) [inline code/datums/emotes.dm]: You skip the cooldown check and the `help_check` with this early return here.

File: code/datums/emotes.dm [hunk 2]
```diff
@@ -22,13 +22,26 @@
...
 /datum/emote/New()
 	if(key_third_person)
 		emote_list[key_third_person] = src
 	mob_type_allowed_typecache = typecacheof(mob_type_allowed_typecache)
 	mob_type_blacklist_typecache = typecacheof(mob_type_blacklist_typecache)
 
+/datum/emote/proc/can_run_robotic_emote(mob/user)
+	if(iscarbon(user))
+		var/obj/item/organ/tongue/T = user.getorganslot("tongue")
+		if(T.status == ORGAN_ROBOTIC)
```
Comment (FlattestGuitar) [inline code/datums/emotes.dm]: `if(T && T.status == ORGAN_ROBOTIC)`

File: code/datums/emotes.dm [hunk 3]
```diff
@@ -22,13 +22,26 @@
...
+			return TRUE
+	if(issilicon(user))
+		return TRUE
+	if(isdrone(user))
+		return TRUE
+	else
```
Comment (FlattestGuitar) [inline code/datums/emotes.dm]: drop the `else`, make the `FALSE` return less indented

**Review by AndrewMontagne — APPROVED** [review]
LGTM

**Pull request merged**